### ISSUE TITLE
Support time unit array

### ISSUE THREAD

**Issue by domoritz** [vega/vega-lite#6351]
Vega supports arrays for interval time units. For example `["year", "month", "date"]`. In Vega-Lite, we support strings `yearmonthdate` since they are easier to type and validate. However, it may make sense to support the array syntax. One limitation of our string approach is that we don't support all combinations of time units (e.g. `yearseconds`). We don't want to just support `string` since we would lose the ability to validate. However, adding the array syntax would introduce another way to specify interval time units.

**Comment by jwoLondon** [comment]
I've always found `yearMonthDateHoursMinutes` etc. rather clumsy so would welcome a more elegant solution. Given that the sequence is always in order of temporal magnitude, why not allow:

- single string or single element array for lowest level (e.g. `"minutes"` or `["quarter"]`)
- 2-element array [_highest_, _lowest_] for multi-level (e.g. `["year","minutes"]` equivalent to the current `yearMonthDateHoursMinutes`)
- 3-element array [_highest_,_lowest_,_step_] for sequence with non-standard step (e.g. `["year","month","month"]` to avoid defaulting to what is currently `yearMonthQuarter`.

**Comment by kanitw** [comment]
> 3-element array [highest,lowest,step] for sequence with non-standard step (e.g. ["year","month","month"] to avoid defaulting to what is currently yearMonthQuarter.

We already support `{unit: string, step: number}` for this, which is a bit more flexible since you can do every 2 months/weeks. 

> 2-element array [highest, lowest] for multi-level (e.g. ["year","minutes"] equivalent to the current yearMonthDateHoursMinutes)
I agree and We thought about this too.

But the tricky part is that we now have some units that aren't a date time component (e.g., quarter, day of year, day of weeks) and thus can't be easily inferred from a two-element array.

**Comment by kanitw** [comment]
That said, for the two element array case, I could see an argument for only filling in standard date components (year/month/date/hours/minutes/seconds/ms) and not the special ones ( (e.g., quarter, day of year, day of weeks).

**Comment by kanitw** [comment]
The tricky part remains what if I want year and hours (but not month, date), it's unclear how to state whether I want the inference in the middle or not.

**Comment by willium** [comment]
> Vega supports arrays for interval time units. For example `["year", "month", "date"]`. In Vega-Lite, we support strings `yearmonthdate` since they are easier to type and validate.
For what it's worth, I don't think it would _that_ hard (albeit tedious in places) to type check all the possible arrays

**Comment by domoritz** [comment]
Nope, validation is not hard at all (as long as you don't care about the order). The point about typing it still stands.

**Comment by willium** [comment]
What am I missing about typing that makes it difficult?
It seems like we would just need to manually itemize each possible tuple

**Comment by domoritz** [comment]
`"yearMonthDateHoursMinutes"` vs `["year", "month", "date", "horus", "minutes"]` (the former will also show autocomplete for the complete string when you start typing it).

**Comment by willium** [comment]
autocomplete — sure, fair point. Though doesn't seem like a good reason to avoid the cleaner syntax.
But why is typing the tuple difficult? It seems the type would just be the literal (same as it is for the string)

**Comment by domoritz** [comment]
Oh, I hadn't thought of typing the whole tuple. I haven't checked whether that approach works with autocomplete in VSCode for TS and the editor with JSON schema. 

Btw, I would be in favor of the array syntax and would be happy to review a pull request.

**Comment by kanitw** [comment]
Array sounds good to me too.
